app/Http/Controllers/Task/TaskController.php

```diff
@@ -44,9 +46,49 @@ public function postAddTasks(TaskRequest $request)
     public function index()
     {
         $list_tasks = $this->tasks->getAllTasks();
-        return view('tasks.list_task',[
-           'list_tasks' => $list_tasks
+        return view('tasks.list_task', [
+            'list_tasks' => $list_tasks
         ], compact('list_tasks'));
     }
 
+    public function getEditTasks($id)
+    {
+        try {
+            $task = $this->tasks->findOrFail($id);
+            $list_users = $this->users->getAllUser();
+            return view('tasks.edit_task', [
+                'idTasks' => $task,
+                'list_users' => $list_users
+            ], compact('idTasks'));
+        } catch (ModelNotFoundException $e) {
+            return redirect()->back();
+        }
+    }
+
+    public function postEditTasks(TaskRequest $request, $id)
+    {
+        try {
+            $task = $this->tasks->findOrFail($id);
```

Comment:
Tìm hiểu và comment phân biệt giúp a `findOrFail()` và `find()` khác gì nhau e nhé. Trả về cái gì khi tìm thấy và k tìm thấy đối tượng
Rồi a củng cố lại chỗ này cho 1 chút

Reply:
- find($id): Nhận một $id và trả về một tập hợp dữ liệu. Nếu không tìm thấy, sẽ trả về null

- findOrFail($id): Nhận một $id và trả về một tâp dữ liệu. Nếu không tìm thấy thì nó sẽ tự throws lỗi.

Reply:
Chỗ này trước a thấy e dùng `find()` rồi mà còn throws lỗi nên a muốn e tìm hiểu thêm về `findOrFail()` để có thể áp dụng sau này.
Chứ trong khi làm việc thì họ sẽ thường dùng `find() `nhiều hơn là `findOrFail()`.  Do cấu trúc sau này với các dự án sẽ k xử lý Logic nhiều ở Controller. 
. find() không tìm thấy sẽ trả về null nên chỉ cần check data trả về nếu có thì mới update, delete. Vì sau này những việc xử lý logic sẽ không viết hết ở Controller mà sẽ đưa qua Repository (Sau này a Quốc có thể sẽ comment e dùng Repository). Controller chỉ gọi lại mà thôi

. `findOrFail()` khi không tìm thấy sẽ trả throws lỗi, nên sẽ phải bắt exception để xử lý như thế nào. Còn khi e dùng ::findOrFail($id) ở Controler thì nếu không tìm thấy nó tự break ngay và trả về trang 404 nếu em không bắt try catch exception.

Còn sử dụng find() thì giống như cách e viết ở Model lúc đầu (Nhưng sau này sẽ viết ở Repository. Sẽ viết 1 hàm update chung (Nhưng chỉ trả về true false) chứ không redirect gì trong này

Ví dụ như này.
```
public function update($id, array $attributes)
    {
        $result = $this->find($id);
        if ($result) {
            $result->update($attributes);
            return $result;
        }

        return false;
    }
```
Params ở đây sẽ là 1 mảng, Tức là truyền vào attributes nào sẽ cập nhật giá trị đó cho task.( Như này sẽ dùng lại được sau này, muốn update cái gì chỉ cần truyền vào cái đó.) Chứ không set kiểu cố định các attribute như lúc e viết ở Model như này. Như này thì cái hàm đó nó chỉ dùng đc có 1 lần.
```
$task->name = $request->name;
$task->description = $request->description;
```

Tức là bên Controller gọi sẽ như thế này. Ở đây ví dụ update tất cả atributes truyền lên từ form

```
public function update(Request $request, $id)
    {
        $data = $request->all(); // Ở đây ví dụ update tất cả Atributes truyền lên từ form

        $task = $this->task->update($id, $data);
        if($task)
          return view('home.task', compact('task'));
        else
         return redirect()->back();
    }

```
Còn giả sử anh chỉ cần update cái `name` thôi thì chỉ cần gọi như này  $task = $this->task->update($id, ['name' =>  $request->name)] );